cmd/daemon/player.go

```diff
@@ -223,12 +223,35 @@ func (p *AppPlayer) handlePlayerCommand(req dealer.RequestPayload) error {
 		p.state.player.Options.RepeatingContext = req.Command.Options.PlayerOptionsOverride.RepeatingContext
 
 		var skipTo skipToFunc
-		if len(req.Command.Options.SkipTo.TrackUri) > 0 || len(req.Command.Options.SkipTo.TrackUid) > 0 {
+		if len(req.Command.Options.SkipTo.TrackUri) > 0 || len(req.Command.Options.SkipTo.TrackUid) > 0 || req.Command.Options.SkipTo.TrackIndex > 0 {
 			skipTo = func(track *connectpb.ContextTrack) bool {
 				if len(req.Command.Options.SkipTo.TrackUid) > 0 && req.Command.Options.SkipTo.TrackUid == track.Uid {
 					return true
 				} else if len(req.Command.Options.SkipTo.TrackUri) > 0 && req.Command.Options.SkipTo.TrackUri == track.Uri {
 					return true
+				} else if req.Command.Options.SkipTo.TrackIndex != 0 {
+					ndxRelative := req.Command.Options.SkipTo.TrackIndex - int(p.state.tracks.Index().Track)
+					var ndxNormalized = -1
+					if ndxRelative < 0 {
+						ndxNormalized += -ndxRelative
+					} else {
+						ndxNormalized += ndxRelative
+					}
+
+					var uri string
+					if ndxRelative < 0 {
+						if len(p.state.tracks.PrevTracks()) <= ndxNormalized {
+							return false
+						}
+						uri = p.state.tracks.PrevTracks()[ndxNormalized].Uri
+					} else {
+						if len(p.state.tracks.NextTracks()) <= ndxNormalized {
+							return false
+						}
+						uri = p.state.tracks.NextTracks()[ndxNormalized].Uri
+					}
+
+					return track.Uri == uri
```

Comment:
This feels a bit clunky, perhaps a better approach would be to just count how many times the callback is called? Or even change the skip function to add an index parameter, along with the track, that is incremented where the function is called, so that all of this becomes a comparison.

Reply:
Yeah you are right, didn't think this would work because of tracks in the queue, but spotify seems to have thought of this also.